.github/actions/update-releases-md/index.js

```diff
@@ -0,0 +1,111 @@
+const actionUtils = require('../action-utils.js');
+
+async function run() {
+    const [core, github] = await actionUtils.installAndRequirePackages("@actions/core", "@actions/github");
+
+    const releasesDataFile = core.getInput("releases_json_file", { required: true });
+    const outputFile = core.getInput("releases_md_file", { required: true });
+
+    const repoOwner = github.context.payload.repository.owner.login;
+    const repoName = github.context.payload.repository.name;
+
+    try {
+        const releasesData = JSON.parse(await actionUtils.readFile(releasesDataFile));
+
+        const releasesMdContent = generateReleasesMdContent(releasesData, repoOwner, repoName);
+
+        await actionUtils.writeFile(outputFile, releasesMdContent);
+    } catch (error) {
+        core.setFailed(error);
+    }
+}
+
+function generateReleasesMdContent(releasesData, repoOwner, repoName) {
+    let supportedReleasesTable = '';
+    let previewReleasesTable = '';
+    let outOfSupportReleasesTable = '';
+
+    for (const releaseKey of releasesData.supported) {
+        supportedReleasesTable += `${generateTableRow(releasesData.releases[releaseKey], repoOwner, repoName, true)}\n`;
+    }
+
+    for (const releaseKey of releasesData.preview) {
```

Comment:
nit: for readability, can we put these loops in the same order as the ones below where we're appending the content?